src/storage/hub.ts

```diff
@@ -51,8 +51,20 @@ export async function uploadToGaiaHub(
     }
   )
   if (!response.ok) {
-    throw new Error('Error when uploading to Gaia hub')
-  } 
+    const responseDescription = await getResponseDescription(response)
+    const errorMsg = `Error when uploading to Gaia hub. ${responseDescription}`
```

Comment:
Gaia returns useful JSON error payloads for many error conditions. The function that provides this `responseDescription` value uses `response.text()`, so the `Error` object thrown will have a message string with embedded stringified json. This makes it hard for blockstack.js and apps to perform condition-specific error handling. 

It would be useful if these Gaia response errors contained the hub error response data. For example, the errors could extend something like:
```ts
interface GaiaHubErrorResponse {
  status: number, 
  statusText: string,
  body?: string | any
}

class GaiaHubError extends BlockstackError {
  hubError: {
    message?: string
    error?: string
    statusCode: number
    statusText: string
    [prop: string]: any
  }

  constructor(error: ErrorType, response: GaiaHubErrorResponse) {
    super(error)
    this.hubError = {
      statusCode: response.status,
      statusText: response.statusText
    }
    if (typeof response.body === 'string') {
      this.hubError.message = response.body
    } else if (typeof response.body === 'object') {
      Object.assign(this.hubError, response.body)
    }
  }
}
```

The `getResponseDescription` function could return the JSON if available, something like:
```ts
async function getGaiaErrorResponse(response: Response): Promise<GaiaHubErrorResponse> {
  let responseMsg = ''
  let responseJson: any | undefined
  try {
    responseMsg = await response.text()
    try {
      responseJson = JSON.parse(responseMsg)
    } catch (error) { 
      // ignore... server did not return JSON, use string response
    }
  } catch (error) {
    Logger.debug(`Error getting bad http response text: ${error}`)
  }
  return {
    status: response.status,
    statusText: response.statusText,
    body: responseJson || responseMsg
  }
}
```

Reply:
Done! Implemented changes based on your suggestions. Thanks!

Reply:
`getBlockstackErrorFromResponse` in `utils.ts` is the central place to process gaia hub errors.